exec/java-exec/src/main/java/org/apache/drill/exec/planner/common/DrillRelOptUtil.java

```diff
@@ -169,4 +176,223 @@ private static boolean containIdentity(List<? extends RexNode> exps,
     }
     return true;
   }
+
+  /**
+   * Copied from {@link RelOptUtil#splitJoinCondition(RelNode, RelNode, RexNode, List, List)}. Modified to rewrite
```

Comment:
I will followup with a JIRA on Calcite project to see if we can push this change to Calcite. 

The function ```RelOptUtil#splitJoinCondition``` in the current form itself seems to have a problem/limitation. Currently it just returns the left and right join key indices, but doesn't return whether the condition is ```EQUAL``` or ```IS NOT DISTINCT FROM``` (it adds the key pair if they have either of these function in comparison).